alibi_detect/utils/loading.py

```diff
@@ -0,0 +1,625 @@
+            return load_detector_legacy(filepath, '.pickle', **kwargs)
+        else:
+            raise ValueError('Neither meta.dill, meta.pickle or config.toml exist in {}.'.format(filepath))
+
+    # No other file types are accepted, so if not dir raise error
+    else:
+        raise ValueError("load_detector accepts only a filepath to a directory, or a config.toml file.")
+
+
+# TODO - will eventually become load_detector
+def _load_detector_config(filepath: Union[str, os.PathLike]) -> Detectors:
+    """
+    Loads a drift detector specified in a detector config dict. Validation is performed with pydantic.
+
+    Parameters
+    ----------
+    filepath
+        Directory containing the `config.toml` file.
+
+    Returns
+    -------
+    The instantiated detector.
+    """
+    # Load toml if needed
+    if isinstance(filepath, (str, os.PathLike)):
+        config_file = Path(deepcopy(filepath))
+        config_dir = config_file.parent
+        cfg = read_config(config_file)
+    else:
+        raise ValueError("`filepath` should point to a directory containing a 'config.toml' file.")
+
+    # Resolve and validate config
+    cfg = validate_config(cfg)
+    logger.info('Validated unresolved config.')
+    cfg = resolve_cfg(cfg, config_dir=config_dir)
+    cfg = validate_config(cfg, resolved=True)
+    logger.info('Validated resolved config.')
+
+    # Backend and detector type
+    detector_name = cfg.get('name')
+    backend = cfg.pop('backend')  # popping so that cfg left as kwargs + `name` when passed to _init_detector
+    if backend.lower() != 'tensorflow':
+        raise NotImplementedError('Loading detectors with PyTorch or sklearn backend is not yet supported.')
+
+    # Get x_ref
+    x_ref = cfg.pop('x_ref')
+    if isinstance(x_ref, str):  # If x_ref still a str, resolving must have failed
+        raise ValueError("Failed to resolve x_ref field.")
+
+    # Get kernel
+    kernel = cfg.pop('kernel', None)  # Don't need to check if None as kernel=None defaults to GaussianRBF
+    if isinstance(kernel, dict):
+        logger.info('Loading kernel.')
+        kernel = _load_kernel_config(kernel, backend, cfg['device'])
+
+    # Get preprocess_fn
+    preprocess_fn = cfg.pop('preprocess_fn')
+    if isinstance(preprocess_fn, dict):
+        logger.info('Loading preprocess_fn.')
+        preprocess_fn = _load_preprocess(preprocess_fn, backend=backend)
+
+    # Get model
+    model = cfg.pop('model', None)
+    if model is not None:
+        if not isinstance(model, SupportedModels):
+            raise ValueError(f"Failed to load the {detector_name}'s model."
+                             "Is the `model` field specified, and is the model a supported type?")
+
+    # Init detector
+    logger.info('Instantiating detector.')
+    detector = _init_detector(x_ref, cfg, preprocess_fn=preprocess_fn, model=model, kernel=kernel, backend=backend)
+
+    # Update metadata
+    detector.meta.update({'config_file': str(config_file.resolve())})
```

Comment:
It seems to me we're not adding this metadata for normally (non-config) instantiated detectors? Also what is the use case of having a plain string in the metadata? Any manipulation would be done by the detector's `get_config` method.

Reply:
This is simply adding the absolute filepath of the `config.toml` file to the metadata, so that the user could see where the instantiated detector came from at runtime. Can remove if you think unnecessary!

Reply:
Not sure if needed as the user wouldn't be able to load a detector if they didn't know the path already? Also, as I mentioned, this introduces a disparity in the metadata between runtime- and config-defined detectors.

Reply:
Good point, I'll remove!

Reply:
p.s. @jklaise, on a related note, we need to add the `version_warning` field to the detector somewhere, so that we have access to it at save time. I currently add this in to `detector.meta`. Do you think this should go somewhere else, i.e. just as `detector.version_warning`? I get the impression you're thinking about us having consistent detector metadata across all use cases etc?